packages/scroller/src/vaadin-scroller.js

```diff
@@ -85,6 +85,30 @@ class Scroller extends FocusMixin(ElementMixin(ThemableMixin(PolymerElement))) {
     };
   }
 
+  /** @protected */
+  ready() {
+    super.ready();
+
+    // Update overflow attribute on resize
+    this.__resizeObserver = new ResizeObserver(() => {
+      this.__updateOverflow();
+    });
+    this.__resizeObserver.observe(this);
+
+    // Update overflow attribute on DOM mutations
+    this.__mutationObserver = new MutationObserver(() => {
+      this.__updateOverflow();
+    });
```

Comment:
Should check the mutation record here, and remove the ResizeObserver from removed nodes and observe added nodes.